contracts/src/test/kotlin/com/contractsAndStates/states/RoadStateTest.kt

```diff
@@ -0,0 +1,179 @@
+                        .with(entry.key)
+                        .with(RollTrigger(3))
+                        .with(entry.key == HexTileType.Desert)
+            }
+        }
+    }
+
+
+    @Test
+    fun `Longest road of 1`() {
+        buildRoads(p1.party, listOf(0 to 1))
+        buildBoard()
+        assertEquals(1, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 2 same hex`() {
+        buildRoads(p1.party, listOf(0 to 1, 0 to 2))
+        buildBoard()
+        assertEquals(2, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 3 same hex`() {
+        buildRoads(p1.party, listOf(0 to 1, 0 to 2, 0 to 0))
+        buildBoard()
+        assertEquals(3, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 6 same hex`() {
+        buildRoads(p1.party, listOf(0 to 0, 0 to 1, 0 to 2, 0 to 3, 0 to 4, 0 to 5))
+        buildBoard()
+        assertEquals(6, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 2 different hex`() {
+        buildRoads(p1.party, listOf(0 to 2, 1 to 3))
+        buildBoard()
+        assertEquals(2, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 3 with split path`() {
+        buildRoads(p1.party, listOf(0 to 2, 1 to 3, 1 to 4, 4 to 1))
+        buildBoard()
+        assertEquals(3, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 8 with loop`() {
+        buildRoads(p1.party, listOf(0 to 0, 0 to 1, 0 to 2, 0 to 3, 0 to 4, 0 to 5, 1 to 3, 1 to 2))
+        buildBoard()
+        assertEquals(8, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `longest road of 13 with 2 loops`() {
+        buildRoads(p1.party, listOf(0 to 0, 0 to 1, 0 to 2, 0 to 3, 0 to 4, 0 to 5, 1 to 3, 1 to 2,
+                                    5 to 0, 5 to 1, 5 to 2, 5 to 3, 5 to 4))
+        buildBoard()
+        assertEquals(13, longestRoadForPlayer(board, roads.toList(), listOf(), p1.party).count())
+    }
+
+    @Test
+    fun `road of 2 with 1 settlement in the middle in the same hex`() {
+        buildRoads(p1.party, listOf(5 to 0, 5 to 1))
+        buildSettlements(p2.party, listOf(5 to 1))
+        buildBoard()
+        assertEquals(1, longestRoadForPlayer(board, roads.toList(), settlements.toList(), p1.party).count())
+    }
+
+    @Test
+    fun `road of 2 with 1 own settlement in the middle`() {
+        buildRoads(p1.party, listOf(5 to 0, 5 to 1))
+        buildSettlements(p1.party, listOf(6 to 5))
+        buildBoard()
+        assertEquals(2, longestRoadForPlayer(board, roads.toList(), settlements.toList(), p1.party).count())
+    }
+}
+
+class AssignLongestRoadTests {
```

Comment:
I would make it a different file, even if what they test is in the same file.

Reply:
Ok. Makes sense.